### PR TITLE
BUGFIX: SD-1045 Metric name is shown as localIdentifier in KD if insight has local date filter

### PR DESCRIPTION
**Pull request by anvuong119**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/3334
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/3093

# Related Jira tasks

SD-1045: https://jira.intgdc.com/browse/SD-1045

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8141/ : SUCCESS in 8m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4441/ : SUCCESS in 4s

**Comment by anvuong119** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1736.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4442/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3669/ : SUCCESS in 19m 03s

**Comment by anvuong119** [comment]
extended test - storybook

**Comment by anvuong119** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8142/ : SUCCESS in 3m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4444/ : SUCCESS in 3s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4268/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4443/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3670/ : SUCCESS in 16m 14s

File: src/helpers/measureTitleHelper.ts [hunk 1]
```diff
@@ -308,6 +308,31 @@ function removeTitleForSimpleMeasure(bucket: IBucket): IBucket {
...
+                return false;
+            }
+            const filters: VisualizationObject.VisualizationObjectFilter[] =
+                definition.measureDefinition.filters;
+            return filters && filters.length > 0 && VisualizationObject.isDateFilter(filters[0]);
+        },
+    );
+}
+
+function isBucketsContainingMeasureDateFilter(buckets: IBucket[]): boolean {
```
Comment (ptquang86) [inline src/helpers/measureTitleHelper.ts:328]: `areBuckets` ?

File: src/helpers/measureTitleHelper.ts [hunk 2]
```diff
@@ -308,6 +308,31 @@ function removeTitleForSimpleMeasure(bucket: IBucket): IBucket {
...
+            if (!VisualizationObject.isMeasure(bucketItem)) {
+                return false;
+            }
+            const definition: IMeasureDefinitionType = bucketItem.measure.definition;
+            if (!VisualizationObject.isMeasureDefinition(definition)) {
+                return false;
+            }
+            const filters: VisualizationObject.VisualizationObjectFilter[] =
+                definition.measureDefinition.filters;
+            return filters && filters.length > 0 && VisualizationObject.isDateFilter(filters[0]);
```
Comment (ptquang86) [inline src/helpers/measureTitleHelper.ts:323]: could date filter have another index (not 0)?

File: src/helpers/measureTitleHelper.ts [hunk 3]
```diff
@@ -308,6 +308,31 @@ function removeTitleForSimpleMeasure(bucket: IBucket): IBucket {
...
+        },
+    );
+}
+
+function isBucketsContainingMeasureDateFilter(buckets: IBucket[]): boolean {
+    return !!buckets.find(
+        (bucket: IBucket): boolean => {
+            return isBucketContainingMeasureDateFilter(bucket);
+        },
+    );
```
Comment (ptquang86) [inline src/helpers/measureTitleHelper.ts:333]: shorter version 
`return !!buckets.find(isBucketContainingMeasureDateFilter)`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8150/ : SUCCESS in 5m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4448/ : SUCCESS in 4s

File: src/helpers/measureTitleHelper.ts [hunk 4]
```diff
@@ -308,6 +308,33 @@ function removeTitleForSimpleMeasure(bucket: IBucket): IBucket {
...
+                return false;
+            }
+            const filters: VisualizationObject.VisualizationObjectFilter[] =
+                definition.measureDefinition.filters;
+            return (
+                filters &&
+                !!filters.find(
+                    (filter: VisualizationObject.VisualizationObjectFilter): boolean =>
+                        VisualizationObject.isDateFilter(filter),
+                )
```
Comment (ptquang86) [inline src/helpers/measureTitleHelper.ts:328]: shorter version
```
!!filters.find(VisualizationObject.isDateFilter)
```

File: src/helpers/measureTitleHelper.ts [hunk 5]
```diff
@@ -323,6 +350,11 @@ function removeTitleForSimpleMeasure(bucket: IBucket): IBucket {
...
     mdObject: IVisualizationObjectContent,
 ): IVisualizationObjectContent {
+    if (areBucketsContainingMeasureDateFilter(mdObject.buckets)) {
+        // If there is one bucket containing a measure with date filter, all other measures are considered adhoc measures
+        // and all the titles should be left intact
+        return mdObject;
+    }
     return {
         ...mdObject,
         buckets: mdObject.buckets.map((bucket: IBucket): IBucket => removeTitleForSimpleMeasure(bucket)),
```
Comment (ptquang86) [inline src/helpers/measureTitleHelper.ts:360]:
```
mdObject.buckets.map(removeTitleForSimpleMeasure),

```

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8152/ : SUCCESS in 4m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4450/ : SUCCESS in 4s

**Comment by anvuong119** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1736.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/4451/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3671/ : SUCCESS in 21m 16s

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/4273/ : SUCCESS in 12m 02s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/192/ : SUCCESS in 14s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/8153/ : SUCCESS in 2m 48s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1971/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/3000/ : SUCCESS in 18m 00s

**Pull request merged**